modules/common/logging/client.nix

```diff
@@ -0,0 +1,53 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  ...
+}: let
+  cfg = config.ghaf.logging.client;
+  endpointUrl = config.ghaf.logging.client.endpoint;
+in {
+  options.ghaf.logging.client.endpoint = lib.mkOption {
+    description = ''
+      Assign endpoint url value to the alloy.service running in
+      different log producers. This endpoint URL will include
+      protocol, upstream, address along with port value.
+    '';
+    type = lib.types.str;
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.etc."alloy/client.alloy" = {
+      text = ''
+        discovery.relabel "journal" {
+          targets = []
+          rule {
+            source_labels = ["__journal__hostname"]
+            target_label  = "nodename"
+          }
+        }
+
+        loki.source.journal "journal" {
+          path          = "/var/log/journal"
+          relabel_rules = discovery.relabel.journal.rules
+          forward_to    = [loki.write.adminvm.receiver]
+        }
+
+        loki.write "adminvm" {
+          endpoint {
+            url = "${endpointUrl}"
+          }
+        }
+      '';
+      # The UNIX file mode bits
+      mode = "0644";
+    };
+
+    services.alloy.enable = true;
+    # Once alloy.service in admin-vm stopped this service will
+    # still keep on retrying to send logs batch, so we need to
+    # stop it forcefully.
+    systemd.services.alloy.serviceConfig.TimeoutStopSec = 0;
```

Comment:
Do we want to give it a second or two to actually have a chance to flush things out when applicable?

Reply:
I propose setting this to 4 seconds like in the other file.

Reply:
Thanks @flokli this is good point. I have kept **0**  because in case of shutdown/reboot I noticed in `admin-vm` alloy.service is getting stopped first and I will wait for 4 sec [here](https://github.com/tiiuae/ghaf/pull/625/files#diff-87d86a1e265f336e25adef2e8f43d3e185ddb94bbf7c68e8e7a59d0befa1c9fcR70) to flush things and log senders to stop, assuming once alloy.service in log-aggregator stopped, does it make sense for clients to wait more ? Since there will be no listener/receiver running it will be only retry happening to send batch during that time and it will never reach target. 
I could be wrong here in terms timing of services are stopped in case of shutdown/reboot.

Reply:
if you shut down both adminvm and a regular "client vm", flushing logs from the "client vm" to the adminvm might not succeed, sure. However, both shutdowns happen at the same time, so using the same timeout as on the admin shouldn't cause an increase in total shutdown time.

If I interpret the manpage right, setting this to 0 does immediately send a `SIGKILL` right after the `SIGTERM`, which is quite brutal. We should give it some time to flush things to the adminvm if possible, clean up open files etc. This code path is also exercised if the "client vm" is simply rebooted.

Reply:
Sure, increasing timeout in client.nix didn't impact on total system shutdown time. So modified code accordingly.